src/markerclusterer.ts

```diff
@@ -178,8 +178,22 @@ export class MarkerClusterer extends OverlayViewSafe {
 
       // allow algorithms to return flag on whether the clusters/markers have changed
       if (changed || changed == undefined) {
+        clusters.forEach((cluster, index) => {
+          // if there's clusters that has been "processed" (Cluster.marker is set), then reuse it instead of reprocessing it
+          const duplicate = this.clusters.findIndex((old) => old.marker && old.marker.getPosition().equals(cluster.position));
```

Comment:
There might be some possible optimization here.

We are searching the former clusters for each new cluster. So basically o(n^2).

Using a map of positionString -> cluster should speed that up.